src/audio/dai-zephyr.c

```diff
@@ -496,8 +496,12 @@ __cold int dai_common_new(struct dai_data *dd, struct comp_dev *dev,
 	dd->ipc_config = *dai_cfg;
 
 	/* request GP LP DMA with shared access privilege */
+#if CONFIG_DAI_VIRTUAL
+	dir = SOF_DMA_DIR_MEM_TO_MEM;
+#else
```

Comment:
Is there anyway we can do this without the ifdef i.e. we may want to be able to combine virtual and real DAIs, you could add a flag to the dai structure to indicate virtual ?